lib/cartopy/mpl/gridliner.py

```diff
@@ -799,17 +811,30 @@ def remove_path_dupes(path):
                                         .transformed(self.axes.transData))
                         if '3.1.0' <= matplotlib.__version__ <= '3.1.2':
                             outline_path = remove_path_dupes(outline_path)
+
+                    # Draw non-inline labels on top of map
+                    is_filled = (
+                        False
+                        if self.xpadding < 0 or self.ypadding < 0 else
+                        True
+                    )
+
                     # Inline must be within the map.
                     if ((lonlat == 'lon' and self.x_inline) or
                             (lonlat == 'lat' and self.y_inline)):
                         # TODO: When Matplotlib clip path works on text, this
                         # clipping can be left to it.
                         if outline_path.contains_point(center):
                             visible = True
-                    # Non-inline must not run through the outline.
-                    elif not outline_path.intersects_path(this_path):
+                    elif not outline_path.intersects_path(this_path,
+                                                          filled=is_filled):
```

Comment:
What about adding the boolean tests directly into the argument here instead of creating another variable?